fix(pie-docs): DSW-2327 content update changes in text input documentation (#1962)
justeattakeaway/pie: apps/pie-docs/src/components/text-input/overview/overview.md

Payment method is only available in trailing content.
Use when a text input requires a card number to be entered, and reflect the card  entered by the user.

{% notification {
  type: "warning",

>   message: "This modifier only exists in design, not in development. The functionality is handled by pillar engineers' if required."

Reviewer: ```suggestion
  message: "This modifier only exists in design, not in development. The functionality can be handled by consumers if required."
```

Author: i am gonna update the other references too 🤝

{% contentLayout %}
  {% contentItem %}
    <h4>Placeholder</h4>

>     <p>Placeholder text in the text input offers the user suggestions or examples for what to enter.</p>

Reviewer: I think the previous message was a bit clearer in what not to do. Can I ask why we want to make this change?

Author: Discussed this with @tingting-lau and we decided to revert the change
